sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -3325,8 +3325,11 @@ object GpuOverrides extends Logging {
     expr[org.apache.spark.sql.execution.ScalarSubquery](
       "Subquery that will return only one row and one column",
       ExprChecks.projectOnly(
-        TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_128,
-        TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_128,
+        (TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_128
+            + TypeSig.ARRAY + TypeSig.MAP + TypeSig.STRUCT)
+            .nested(TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_128 +
+                        TypeSig.ARRAY + TypeSig.MAP + TypeSig.STRUCT),
```

Comment:
Nit: I think this can be simplified.
```suggestion
        (TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_128
            + TypeSig.ARRAY + TypeSig.MAP + TypeSig.STRUCT).nested(),
```

Reply:
Done